### PR TITLE
Libgupdate

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

Updates LibG to new versions which contain instancing implementation for primitives, also fixes some bugs the tests helped me find.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Updating LibG so primitives start instancing when rendering.

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/VisualizationTests/HelixWatch3DViewModelTests.cs [hunk 1]
```diff
@@ -1277,7 +1277,6 @@ public void InstancedLinesAreAddedToBackGroundPreviewForEachMatrix()
 
         }
         [Test]
-        [Category("Failure")]
```
Comment (saintentropy) [inline src/VisualizationTests/HelixWatch3DViewModelTests.cs:1280]: Good memory

**Comment by aparajit-pratap** [comment]
@mjkkirschner does this also contain the [bug fix](https://git.autodesk.com/Dynamo/LibG/pull/1299)?

**Comment by mjkkirschner** [comment]
@aparajit-pratap yes I believe it does.
I have to fix some test failures here though so if you want to merge that separately please don't let me hold you up.

File: src/DynamoCoreWpf/Rendering/HelixRenderPackage.cs [hunk 2]
```diff
@@ -588,9 +588,46 @@ public void AddLabel(string label, VertexType vertexType, int index)
...
+                case VertexType.LineInstance:
+                    //
+                    if (vertIndex > Lines.Positions.Count) { return; }
+                    position = Lines.Positions[vertIndex - 1];
+                    break;
+            }
+            if (instanceTransforms.ContainsKey(BaseTessellationId) && instanceIndex <= instanceTransforms[BaseTessellationId].Count)
+            {
+                var transform = instanceTransforms[BaseTessellationId]?[instanceIndex - 1];
+                var transformedLabelPos = transform?.ToMatrix3D().Transform(position.ToPoint3D()).ToVector3();
```
Comment (saintentropy) [inline src/DynamoCoreWpf/Rendering/HelixRenderPackage.cs:623]: Any idea how long this takes.  I guess I am curious perf vs within the AddInstance call where you have access to position info directly.  This is also probably very fast.

**Comment by aparajit-pratap** [comment]
> @aparajit-pratap yes I believe it does. I have to fix some test failures here though so if you want to merge that separately please don't let me hold you up.

@mjkkirschner that's fine, I'll wait, not urgent.

File: src/DynamoCoreWpf/Rendering/HelixRenderPackage.cs (hunk 2 above)
Comment (mjkkirschner) [reply src/DynamoCoreWpf/Rendering/HelixRenderPackage.cs:623]: @saintentropy testing with a stopwatch this appears to take 30 - 50 ms for 125000 labels.

Comment (mjkkirschner) [reply src/DynamoCoreWpf/Rendering/HelixRenderPackage.cs:623]: Screen Shot 2022-05-23 at 10 49 02 AM

**Review by saintentropy — APPROVED** [review]
LGTM

File: src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs [hunk 3]
```diff
@@ -240,23 +241,56 @@ private void GetTessellationDataFromGraphicItem(Guid outputPortId, IGraphicItem
                         if (factory.TessellationParameters.ShowEdges)
                         {
                             //TODO if we start to instance more types, expand this edge generation.
+                            //and the swtich case below.
                             if (graphicItem is Topology topology)
                             {
-                                var edges = topology.Edges;
```
Comment (aparajit-pratap) [inline src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs:245]: It looks like previously `edges` was non-null for any topology whereas now it is non-null for only cuboids?

File: src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs [hunk 4]
```diff
@@ -240,23 +241,56 @@ private void GetTessellationDataFromGraphicItem(Guid outputPortId, IGraphicItem
...
+                    {
+                        if (package.MeshVertexCount > 0)
+                        {
+                            packageWithLabelInstances.AddInstancedLabel(labelKey,
+                                VertexType.MeshInstance,
+                                package.MeshVertexCount, 
+                                packageWithLabelInstances.InstanceCount(instanceableItem.BaseTessellationGuid),
+                                instanceableItem.BaseTessellationGuid) ;
+                        }
+                        else if (package.LineVertexCount > 0)
```
Comment (aparajit-pratap) [inline src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs:285]: What about `PointVertexCount`?

File: src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs (hunk 3 above)
Comment (mjkkirschner) [reply src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs:245]: cuboids are the only solid which can be instanced currently.

Comment (aparajit-pratap) [reply src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs:245]: What about spheres? I thought you have spheres in the screenshot?

Comment (aparajit-pratap) [reply src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs:245]: Scratch that, spheres don't have edges in our case.

**Pull request merged**